EAT-56326: method to check if chart exists in database

duttiv opened the pull request:
If XJog has previously returned null but this check reveals the chart exists, eCom can now throw err0r

duttiv committed:
EAT-56326: Added a method for checking if chart exists in database

duttiv committed:
EAT-56326: Removed unnecessary logging

duttiv committed:
EAT-56326: Removed unused import

duttiv commented on packages/core-pg/src/PostgresPersistenceAdapter.ts:
```diff
@@ -367,6 +367,24 @@ export class PostgresPersistenceAdapter extends PersistenceAdapter<PoolClient> {
     );
   }
 
+  protected async chartExists(
```
I tried to make it similar to `isActivityRegistered` method

martvalja commented on packages/core/src/XJog.ts:
```diff
@@ -304,6 +304,31 @@ export class XJog extends XJogLogEmitter {
+
+    if (!chartIdentifier) {
+      this.trace({ in: 'verifyChart', ref }, 'Failed to parse reference');
+      return null;
```
Maybe return false instead so this only returns booleans? Or maybe even throw/reject promise 🤷‍♂️

duttiv replied:
Decided to just return false, as it seemed cleaner

martvalja reviewed (approved):
One question, otherwise looking good

martvalja reviewed (changes requested):
Ah, bump versions as well for the necessary packages (packages that were changed + packages that depend on those.

Guide: https://github.com/telia-oss/xjog/blob/main/contributing/development.md#versioning-and-publishing

duttiv committed:
EAT-56326: Return false instead of null when chart identifier does not exist

duttiv commented:
> Ah, bump versions as well for the necessary packages (packages that were changed + packages that depend on those.
> 
> Guide: https://github.com/telia-oss/xjog/blob/main/contributing/development.md#versioning-and-publishing

Ouch, ignore for now because I somehow forgot to do the thing I was most anxious about. I'll update version tomorrow and then request the review again

martvalja committed:
Publish

 - @samihult/xjog@0.0.314
 - @samihult/xjog-core-persistence@0.0.36
 - @samihult/xjog-core-pg@0.0.60
 - @samihult/xjog-digest-persistence@0.0.22
 - @samihult/xjog-digest-pg@0.0.115
 - @samihult/xjog-digest-reader@0.0.27
 - @samihult/xjog-digest-writer@0.0.14
 - @samihult/xjog-journal-persistence@0.0.50
 - @samihult/xjog-journal-pg@0.0.114
 - @samihult/xjog-journal-reader@0.0.52
 - @samihult/xjog-journal-writer@0.0.68
 - @samihult/xjog-util@0.0.31
